code/modules/mob/living/carbon/xenomorph/castes/king/abilities_king.dm

```diff
@@ -32,104 +32,266 @@
+#define ZEROFORM_TICK_RATE 0.3 SECONDS
+/datum/action/xeno_action/zero_form_beam
+	name = "Zero-Form Energy Beam"
+	action_icon_state = "zero_form_beam"
+	mechanics_text = "After a windup, concentrates the hives energy into a forward-facing beam that pierces everything, but only hurts living beings."
+	ability_name = "zero form energy beam"
+	plasma_cost = 25
+	cooldown_timer = 10 SECONDS
+	keybind_flags = XACT_KEYBIND_USE_ABILITY
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_ZEROFORMBEAM,
+	)
+	///list of turfs we are hitting while shooting our beam
+	var/list/turf/targets
+	///ref to beam that is currently active
+	var/datum/beam/beam
+	///particle holder for the particle visual effects
+	var/obj/effect/abstract/particle_holder/particles
+	// sound loop for the looping sound lazor beam
+	var/datum/looping_sound/zero_form_beam/sound_loop
+	///ref to looping timer for the fire loop
+	var/timer_ref
+
+/datum/action/xeno_action/zero_form_beam/Destroy()
+	QDEL_NULL(sound_loop)
 	return ..()
 
-/datum/action/xeno_action/activable/gravity_crush/can_use_ability(atom/A, silent, override_flags)
+/datum/action/xeno_action/zero_form_beam/New(Target)
+	. = ..()
+	sound_loop = new
+
+/obj/effect/ebeam/zeroform/Initialize()
+	. = ..()
+	alpha = 0
+	animate(src, alpha = 255, time = ZEROFORM_CHARGE_TIME)
+
+/datum/action/xeno_action/zero_form_beam/can_use_action(silent, override_flags)
 	. = ..()
 	if(!.)
 		return
-	if(!line_of_sight(owner, A, king_crush_dist))
+	if(SSmonitor.gamestate == SHUTTERS_CLOSED && is_ground_level(owner.z))
 		if(!silent)
-			to_chat(owner, span_warning("We must get closer to crush, our mind cannot reach this far."))
+			owner.balloon_alert("too early")
 		return FALSE
 
-/datum/action/xeno_action/activable/gravity_crush/use_ability(atom/A)
-	owner.face_atom(A) //Face towards the target so we don't look silly
-	var/list/turfs = RANGE_TURFS(1, A)
-	playsound(A, 'sound/effects/bomb_fall.ogg', 50, FALSE)
-	apply_filters(turfs)
-	if(!do_after(owner, WINDUP_GRAV, FALSE, owner, BUSY_ICON_DANGER))
-		remove_all_filters()
-		//Consume plasma when cancelling
-		succeed_activate()
+/datum/action/xeno_action/zero_form_beam/action_activate()
+	if(timer_ref)
+		stop_beaming()
+		return
+
+	var/turf/check_turf = get_step(owner, owner.dir)
+	LAZYINITLIST(targets)
+	while(check_turf && length(targets) < ZEROFORM_BEAM_RANGE)
+		targets += check_turf
+		check_turf = get_step(check_turf, owner.dir)
+	if(!LAZYLEN(targets))
+		return
+
+	var/particles_type
+	switch(owner.dir)
+		if(WEST)
+			particles_type = /particles/zero_form/west
+		if(EAST)
+			particles_type = /particles/zero_form/east
+		if(NORTH)
+			particles_type = /particles/zero_form/north
+		if(SOUTH)
+			particles_type = /particles/zero_form/south
+		else
+			particles_type = /particles/zero_form
+	particles = new(owner, particles_type)
+	beam = owner.beam(targets[length(targets)], "plasmabeam", beam_type = /obj/effect/ebeam/zeroform)
```

Comment:
Because xeno sprites are offset by 16, the beam is not straight when shooting north or south. Just needs a pixel_x = 16 to straighten it.